homeassistant/components/geofency/__init__.py

```diff
@@ -129,6 +136,7 @@ def _set_location(hass, data, location_name):
 async def async_unload_entry(hass, entry):
     """Unload a config entry."""
     hass.components.webhook.async_unregister(entry.data[CONF_WEBHOOK_ID])
+    hass.data[DATA_KEY]()
```

Comment:
Why move this from the platform to the main component? It makes more sense to me that the platform owns this data and therefore is responsible for unloading it.

Reply:
There's no entry unload in platforms. This is how config entry components with platforms work.

Reply:
😕 Why do we forward the entry_unload on the next line?

Reply:
To unload the platform. But this is all taken care of by the entity component and platform helper. There's no entry unload function in the platform that will be called.

Reply:
I see. That makes sense. Thanks for the clarification.